docs/standard/serialization/binaryformatter-migration-guide/winforms-migration-guide.md

```diff
@@ -0,0 +1,82 @@
+
+## Drag/Drop
+
+If your drag/drop scenario involves unsupported types, BinaryFormatter would be used when [`Control.DoDragDrop`](https://learn.microsoft.com/dotnet/api/system.windows.forms.control.dodragdrop) is called, and the item is dragged out of process and when DataObject.GetData is called to retrieve your type that has been dragged out of process. With the BinaryFormatter removal, developers will now see a string about BinaryFormatter being removed upon dropping the dragged item in another process when a drag operation starts with unsupported types.
+
+## The WinForms Designer
+
+The WinForms designer also uses BinaryFormatter internally for ResX serialization/deserialization and CodeDom.
+
+Types and properties may be participating in serialization without developers realizing due to the standard behavior of the WinForms designer. One way that BinaryFormatter is used where developers may not be aware of is when a property on a UserControl is introduced and that control is already populated at design-time, then it’s very likely that that data gets serialized into code or resource files. If all the following are true:
+
+- A property contains data at the time when a Form in the Designer gets saved
+- that property is not attributed with `DesignerSerializationVisibility(false)`
+- that property does not have a DefaultValueAttribute
+- that property does not have a respective bool ShouldSerialize\[PropertyName\] method returning false
+
+the Designer looks if that property’s type has a type converter. If it does then it uses it to serialize the property content, otherwise BinaryFormatter is used to serialize the content into the resource file.
+WinForms has added analyzers along with code fixes to help bring awareness to this type of behavior where BinaryFormatter serialization may be occurring without the developer’s knowledge.
+
+# Migrating Away from BinaryFormatter
+
+If unsupported types are used in the affected scenarios, action will need to be taken to complete migration to .NET 9+.
+
+## Clipboard and Drag/Drop
+
+For unsupported types that are used in clipboard and drag/drop operations, it is recommended developers format those types as a byte[] or string payload before passing the data to clipboard or drag/drop APIs. Using JSON is one way to achieve this. Note that adjustments will need to be made to handle receiving a JSON formatted type just as adjustments have been made to place JSON formatted types on clipboard or drag/drop operations. For more information on how to serialize/deserialize the type with JSON, see [How to write .NET objects as JSON (serialize)](https://learn.microsoft.com/dotnet/standard/serialization/system-text-json/how-to)
```

Comment:
Idea: We could introduce extension methods maybe for converting BitMaps or other formats to JSon and back? Maybe that could be also something for the CommunityPack, so we can deliver those things out of band?

Reply:
We handle Bitmaps already without BinaryFormatter. Plan in .NET 10 is to introduce methods that will take care of converting to items to/from JSON. For .NET 9, this will need to be done by developer.